ssz/hash_tree.py

```diff
@@ -0,0 +1,394 @@
+from functools import partial
+import itertools
+from numbers import Integral
+from typing import Any, Generator, Iterable, Optional, Union
+
+from eth_typing import Hash32
+from eth_utils.toolz import drop, iterate, partition, pipe, take
+from pyrsistent import pmap, pvector
+from pyrsistent._transformations import transform
```

Comment:
Unfortunately it's private at the moment. I don't see a reasonable way around using it though and it is a rather important function (maybe not for the hash tree, but definitely for the `Hashable*` objects I'm implementing at the moment -- it allows us to write `new_state = state.transform(("validators", 10, "balance"), 0)` instead of `new_state = state.set("validators", state.validators.set(10, state.validators[10].set("balance", 0)))`

I've opened an [issue](https://github.com/tobgu/pyrsistent/issues/180) about this upstream.

Reply:
Maybe good to make a comment at the site of import and link to that issue for reference.

Reply:
Added a comment there and also pinned the version in the `setup.py` to be safe (with a comment).